gui/wxpython/main_window/frame.py

```diff
@@ -136,92 +141,28 @@ def show_menu_errors(messages):
         self._menuTreeBuilder = LayerManagerMenuData(message_handler=add_menu_error)
         # the search tree and command console
         self._moduleTreeBuilder = LayerManagerModuleTree(message_handler=add_menu_error)
-        self._auimgr = wx.aui.AuiManager(self)
+        self._auimgr = aui.AuiManager(self)
 
         # list of open dialogs
         self.dialogs = dict()
         self.dialogs["preferences"] = None
         self.dialogs["nvizPreferences"] = None
         self.dialogs["atm"] = list()
 
-        # create widgets
-        self._createMenuBar()
-        self.statusbar = self.CreateStatusBar(number=1)
-        self.notebook = self._createNotebook()
-        self._createDataCatalog(self.notebook)
-        self._createDisplay(self.notebook)
-        self._createSearchModule(self.notebook)
-        self._createConsole(self.notebook)
-        self._createPythonShell(self.notebook)
-        self._addPagesToNotebook()
-        self.toolbars = {
-            "workspace": LMWorkspaceToolbar(parent=self),
-            "tools": LMToolsToolbar(parent=self),
-            "misc": LMMiscToolbar(parent=self),
-            "nviz": LMNvizToolbar(parent=self),
-        }
-        self._toolbarsData = {
-            "workspace": (
-                "toolbarWorkspace",  # name
-                _("Workspace Toolbar"),  # caption
-                1,
-                0,
-            ),  # row, position
-            "tools": ("toolbarTools", _("Tools Toolbar"), 1, 1),
-            "misc": ("toolbarMisc", _("Misc Toolbar"), 1, 2),
-            "nviz": ("toolbarNviz", _("3D view Toolbar"), 1, 3),
-        }
-        toolbarsList = ("workspace", "tools", "misc", "nviz")
-        for toolbar in toolbarsList:
-            name, caption, row, position = self._toolbarsData[toolbar]
-            self._auimgr.AddPane(
-                self.toolbars[toolbar],
-                wx.aui.AuiPaneInfo()
-                .Name(name)
-                .Caption(caption)
-                .ToolbarPane()
-                .Top()
-                .Row(row)
-                .Position(position)
-                .LeftDockable(False)
-                .RightDockable(False)
-                .BottomDockable(False)
-                .TopDockable(True)
-                .CloseButton(False)
-                .Layer(2)
-                .BestSize((self.toolbars[toolbar].GetBestSize())),
-            )
+        # set pane sizes according to the full screen size
+        self.PANE_BEST_SIZE = tuple(t / 5 for t in wx.DisplaySize())
```

Comment:
There is a note warning about using this function in the [documentation](https://wxpython.org/Phoenix/docs/html/wx.functions.html#wx.DisplaySize), so I suggest to follow it.

Reply:
I am not sure if [0] really means primary, it is not written anywhere... if not it could be possible to determine which monitor it is according to GetPosition, but unfortunately, I do not have the second monitor to try it out.

Reply:
mmm.. actually, neither on them work (neither wx.DisplaySize nor wx.Display(0).GetGeometry().Get(Size)... I need to have self.Maximize(True) to extend GRASS on the whole screen. I will have a look on it again tomorrow.